lib/puppet/daemon.rb

```diff
@@ -189,7 +176,7 @@ def run_event_loop
     end
 
     reparse_run.disable if Puppet[:filetimeout] == 0
-    agent_run.disable unless agent
+    agent_run.disable unless @agent
```

Comment:
If we always have an agent now as we request it in constructor, do we still need the `unless` check?

Reply:
No. It should've been removed also. I somehow missed it after moving an error raise for missing agent in the constructor from one of the methods.